schemas/json/block.json

```diff
@@ -482,6 +482,100 @@
 					}
 				}
 			]
+		},
+		"variations": {
+			"type": "array",
+			"description": "Block Variations is the API that allows a block to have similar versions of it, but all these versions share some common functionality.",
+			"items": {
+				"type": "object",
+				"required": [ "name", "title" ],
+				"additionalProperties": false,
+				"properties": {
+					"name": {
+						"type": "string",
+						"description": "The unique and machine-readable name."
+					},
+					"title": {
+						"type": "string",
+						"description": "A human-readable variation title."
+					},
+					"description": {
+						"type": "string",
+						"description": "A detailed variation description."
+					},
+					"category": {
+						"description": "A category classification, used in search interfaces to arrange block types by category.",
+						"anyOf": [
+							{
+								"type": "string"
+							},
+							{
+								"enum": [
+									"text",
+									"media",
+									"design",
+									"widgets",
+									"theme",
+									"embed"
+								]
+							}
+						]
+					},
+					"icon": {
+						"description": "An icon helping to visualize the variation. It can have the same shape as the block type.",
+						"oneOf": [
+							{
+								"type": "string"
+							},
+							{
+								"type": "object"
```

Comment:
Do you have an example of the `object` type? I think we can only support the string name of the icons here.

Reply:
It can be multiple things looking like an object on the client in JavaScript, but in `block.json` it's only a string, see the same fields for the block type:

https://github.com/WordPress/gutenberg/blob/1e170ed680572338f44aae98a038e3e120fab97e/schemas/json/block.json#L76-L79

Reply:
Thanks for that context :) Updated in [e70e99e](https://github.com/WordPress/gutenberg/pull/42539/commits/e70e99e523b4c33b2de7c3c4690228861d2c3931)